hydrolib/core/io/net/models.py

```diff
@@ -367,25 +369,167 @@ def __init__(
+            max_dist_to_struc (float): The maximum distance from a node to a structure.
+            limits (List[float]): The limits.
+
+        Returns:
+            List[float]: A list with the updated limits.
+        """
+
+        additional = []
+
+        # Skip the first and the last, these are no structures
+        for i in range(1, len(limits) - 1):
+            # if the distance between two limits is large than twice the max distance to structure,
+            # the mesh point will be too far away. Add a limit on the minimum of half the length and
+            # two times the max distance
+            dist_to_prev_limit = limits[i] - (
+                max(additional[-1], limits[i - 1]) if any(additional) else limits[i - 1]
+            )
+            if dist_to_prev_limit > 2 * max_dist_to_struc:
+                additional.append(
+                    limits[i] - min(2 * max_dist_to_struc, dist_to_prev_limit / 2)
+                )
+
+            dist_to_next_limit = limits[i + 1] - limits[i]
+            if dist_to_next_limit > 2 * max_dist_to_struc:
+                additional.append(
+                    limits[i] + min(2 * max_dist_to_struc, dist_to_next_limit / 2)
+                )
+
+        # Join the limits
+        return sorted(limits + additional)
+
+    def _add_nodes_to_segments(
+        self,
+        offsets: np.ndarray,
+        anchor_pts: List[float],
+        limits: List[float],
+        mesh1d_edge_length: float,
+    ) -> np.ndarray:
+        """Add nodes to segments that are missing a mesh node.
+
+        Args:
+            offsets (np.ndarray): The branch offsets.
+            anchor_pts (List[float]): The anchor points.
+            limits (List[float]): The limits.
+            mesh1d_edge_length (float): The edge length of the 1d mesh.
+
+        Returns:
+            np.ndarray: The array with branch offsets.
+        """
+        # Get upper and lower limits
+        upper_limits = limits[1:]
+        lower_limits = limits[:-1]
+
+        # Determine the segments that are missing a mesh node
+        # Anchor points are added on these segments, such that they will get a mesh node
+        in_range = [
+            ((offsets > lower) & (offsets < upper)).any()
+            for lower, upper in zip(lower_limits, upper_limits)
+        ]
+
+        while not all(in_range):
+            # Get the index of the first segment without grid point
+            i = in_range.index(False)
+
+            # Add it to the anchor pts
+            anchor_pts.append((lower_limits[i] + upper_limits[i]) / 2.0)
+            anchor_pts = sorted(anchor_pts)
+
+            # Generate new offsets
+            offsets = self._generate_1d_spacing(anchor_pts, mesh1d_edge_length)
+
+            # Determine the segments that are missing a grid point
+            in_range = [
+                ((offsets > lower) & (offsets < upper)).any()
+                for lower, upper in zip(lower_limits, upper_limits)
+            ]
```

Comment:
Same as above, might be good to put this in a separate method?

Reply:
Done